serde/src/ser/mod.rs

```diff
@@ -41,7 +41,7 @@ pub trait Serialize {
 ///////////////////////////////////////////////////////////////////////////////
 
 /// A trait that describes a type that can serialize a stream of values into the underlying format.
-pub trait Serializer {
+pub trait Serializer: Sized {
```

Comment:
well... if you look at all the other default method impls, none call methods on the value. I'm aware of the motivation, that formats not caring about tags don't need to implement a new method. But maybe we need something along the lines of `MapVisitor` and `SeqVisitor`.